app/api/event_invoices.py

```diff
@@ -0,0 +1,116 @@
+from app.api.helpers.permissions import jwt_required, is_user_itself, is_admin, is_super_admin
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+from app.models import db
+from app.models.event import Event
+from app.models.user import User
+from app.models.event_invoice import EventInvoice
+from app.models.discount_code import DiscountCode
+
+
+class EventInvoiceSchema(Schema):
+
+    class Meta:
+        type_ = 'event_invoice'
+        self_view = 'v1.event_invoice_detail'
+        self_view_kwargs = {'id': '<id>'}
+        self_view_many = 'v1.event_invoice_list'
+
+    id = fields.Str(dump_only=True)
+    identifier = fields.Str()
+    amount = fields.Float()
+    address = fields.Str()
+    city = fields.Str()
+    state = fields.Str()
+    country = fields.Str()
+    zipcode = fields.Str()
+    created_at = fields.DateTime()
+    completed_at = fields.DateTime(default=None)
+    transaction_id = fields.Str()
+    paid_via = fields.Str()
+    payment_mode = fields.Str()
+    brand = fields.Str()
+    exp_month = fields.Integer()
+    exp_year = fields.Integer()
+    last4 = fields.Str()
+    stripe_token = fields.Str()
+    paypal_token = fields.Str()
+    status = fields.Str()
+    user = Relationship(attribute='user',
+                        self_view='v1.event_invoice_user',
+                        self_view_kwargs={'id': '<id>'},
+                        related_view='v1.user_detail',
+                        related_view_kwargs={'event_invoice_id': '<id>'},
+                        schema='UserSchema',
+                        type_='user')
+    event = Relationship(attribute='event',
+                         self_view='v1.event_invoice_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'event_invoice_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+    discount_code = Relationship(attribute='discount_code',
+                         self_view='v1.event_invoice_discount_code',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.discount_code_detail',
+                         related_view_kwargs={'event_invoice_id': '<id>'},
+                         schema='DiscountCodeSchema',
+                         type_='discount_code')
+
+
+class EventInvoiceDetail(ResourceDetail):
```

Comment:
Please keep the functions in the order List, Detail, Relationship to maintain uniformity in code.